File: api/src/main/java/com/hirrao/javaee/utils/BloodPressureUtil.java
package com.hirrao.javaee.utils;

public class BloodPressureUtil {
    public static String generateClassification(float sbp, float dbp) {
        if (sbp >= 180 || dbp >= 110) {
            return "3级高血压";
        } else if (sbp >= 160 || dbp >= 100) {
            return "2级高血压";
        } else if (sbp >= 140 || dbp >= 90) {
            return "1级高血压";
        } else {
            return "正常";
        }
    }

    public static String generateRiskLevel(float sbp, float dbp) {
        if (sbp >= 180 || dbp >= 110) {
            return "重度";
        } else if (sbp >= 160 || dbp >= 100) {
            return "中度";
        } else if (sbp >= 140 || dbp >= 90) {
            return "轻度";
        } else if (sbp >= 120 || dbp >= 80) {
            return "正常高值";
        } else if (sbp <= 90 || dbp <= 60) {
            return "偏低";
        } else {
            return "正常";
        }
    }
}

File: api/src/main/kotlin/com/hirrao/javaee/utils/BloodPressureUtil.kt
package com.hirrao.javaee.utils

object BloodPressureUtil {

    fun generateClassification(sbp: Float, dbp: Float): String {
        return when {
            sbp >= 180 || dbp >= 110 -> "3级高血压"
            sbp >= 160 || dbp >= 100 -> "2级高血压"
            sbp >= 140 || dbp >= 90 -> "1级高血压"
            else -> "正常"
        }
    }


    fun generateRiskLevel(sbp: Float, dbp: Float): String {
        return when {
            sbp >= 180 || dbp >= 110 -> "重度"
            sbp >= 160 || dbp >= 100 -> "中度"
            sbp >= 140 || dbp >= 90 -> "轻度"
            sbp >= 120 || dbp >= 80 -> "正常高值"
            sbp <= 90 || dbp <= 60 -> "偏低"
            else -> "正常"
        }
    }
}
